examples/sphere/shallow_water.jl

```diff
@@ -66,132 +104,61 @@ end
-const test_name = get(ARGS, 1, "steady_state") # default test case to run
-const test_angle_name = get(ARGS, 2, "alpha0") # default test case to run
-const steady_state_test_name = "steady_state"
-const steady_state_compact_test_name = "steady_state_compact"
-const mountain_test_name = "mountain"
-const rossby_haurwitz_test_name = "rossby_haurwitz"
-const barotropic_instability_test_name = "barotropic_instability"
-const alpha0_test_name = "alpha0"
-const alpha30_test_name = "alpha30"
-const alpha45_test_name = "alpha45"
-const alpha60_test_name = "alpha60"
-
-# Test-specific physical parameters
-if test_angle_name == alpha30_test_name
-    const α = 30.0
-elseif test_angle_name == alpha45_test_name
-    const α = 45.0
-elseif test_angle_name == alpha60_test_name
-    const α = 60.0
-else # default test case, α = 0.0
-    const α = 0.0
-end
-
-if test_name == mountain_test_name
-    const u0 = 20.0
-    const h0 = 5960
-    const a = 20.0 # radius of conical mountain
-    const λc = 90.0 # center of mountain long coord, shifted by 180 compared to the paper, because our λ ∈ [-180, 180] (in the paper it was 270, with λ ∈ [0, 360])
-    const ϕc = 30.0 # center of mountain lat coord
-    const h_s0 = 2e3
-elseif test_name == rossby_haurwitz_test_name
-    const a = 4.0
-    const h0 = 8.0e3
-    const ω = 7.848e-6
-    const K = 7.848e-6
-elseif test_name == steady_state_compact_test_name
-    const u0 = 2 * pi * R / (12 * 86400)
-    const h0 = 2.94e4 / g
-    const ϕᵦ = -30.0
-    const ϕₑ = 90.0
-    const xₑ = 0.3
-elseif test_name == barotropic_instability_test_name
-    const u_max = 80.0
-    const xₑ = 0.3
-    const αₚ = 19.09859
-    const βₚ = 3.81971
-    const h0 = 10158.18617 # value for initial height from Tempest https://github.com/paullric/tempestmodel/blob/master/test/shallowwater_sphere/BarotropicInstabilityTest.cpp#L86
-    const h_hat = 120.0
-    const ϕ₀ = 25.71428
-    const ϕ₁ = 64.28571
-    const ϕ₂ = 45.0
-    const eₙ = exp(-4.0 / (deg2rad(ϕ₁) - deg2rad(ϕ₀))^2)
-else # default case, steady-state test case
-    const u0 = 2 * pi * R / (12 * 86400)
-    const h0 = 2.94e4 / g
-end
-
-# Set up discretization
-ne = 9 # the rossby_haurwitz test case's initial state has a singularity at the pole. We avoid it by using odd number of elements
-Nq = 4
-
-domain = Domains.SphereDomain(R)
-mesh = Meshes.EquiangularCubedSphere(domain, ne)
-quad = Spaces.Quadratures.GLL{Nq}()
-if usempi
-    grid_topology = Topologies.DistributedTopology2D(context, mesh)
-    global_grid_topology = Topologies.Topology2D(mesh)
-    space = Spaces.SpectralElementSpace2D(grid_topology, quad)
-    global_space = Spaces.SpectralElementSpace2D(global_grid_topology, quad)
-else
-    grid_topology = Topologies.Topology2D(mesh)
-    global_space = space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+Base.@kwdef struct BarotropicInstabilityTest{FT, P} <: AbstractTest
+    params::P = Params{FT}()
+    u_max::FT = 80.0
+    xₑ::FT = 0.3
```

Comment:
Please only include this in the structs if the problem requires it ; (e.g. Galewsky test case doesn't need it but Rossby-Haurwitz does )